File: test_files/typescript/advanced_autocomplete.ts
type InvoiceOptions = {
  taxRate: number;
  currency: "USD" | "EUR";
  note?: string;
};

type LineItem = {
  sku: string;
  price: number;
  quantity: number;
};

function buildInvoice(customerId: string, items: LineItem[], options: InvoiceOptions): string {
  const subtotal = items.reduce((sum, item) => sum + item.price * item.quantity, 0);
  const total = subtotal * (1 + options.taxRate);
  return `${customerId}:${options.currency}:${total.toFixed(2)}`;
}

export function runObjectArgCase(items: LineItem[]): string {
  const invoice = buildInvoice("C-100", items, 
  return invoice;
}

export function runTemplateLiteralCase(invoice: { id: string; total: number }): string {
  const label = `Invoice ${invoice.id
  return label;
}

export function runChainCase(lines: string[]): string {
  const first = lines.map((line) => line.trim()).filter(Boolean).join(", ").split(
  return first;
}

File: test_files/java/AdvancedAutocomplete.java
import java.util.List;
import java.util.stream.Collectors;

public class AdvancedAutocomplete {
    record Invoice(String id, double total) {}

    static String buildSummary(String customerId, List<Double> amounts, double taxRate, String currency) {
        double subtotal = amounts.stream().mapToDouble(Double::doubleValue).sum();
        double total = subtotal * (1.0 + taxRate);
        return customerId + ":" + currency + ":" + String.format("%.2f", total);
    }

    static String runArgumentsCase(List<Double> amounts) {
        String summary = buildSummary("C-100", amounts, 
        return summary;
    }

    static String runFormatCase(Invoice invoice) {
        String label = String.format("Invoice %s", invoice.id(
        return label;
    }

    static String runStreamCase(List<String> lines) {
        String joined = lines.stream().map(String::trim).filter(value -> !value.isEmpty()).collect(
        return joined;
    }
}
